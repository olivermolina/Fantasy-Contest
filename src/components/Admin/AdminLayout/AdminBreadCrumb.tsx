import React from 'react';
import Link from '@mui/material/Link';
import HomeIcon from '@mui/icons-material/Home';
import NavigateNextIcon from '@mui/icons-material/NavigateNext';
import Breadcrumbs from '@mui/material/Breadcrumbs';
import { useRouter } from 'next/router';
import { UrlPaths } from '~/constants/UrlPaths';

const breadcrumbNameMap: { [key: string]: string } = {
  [UrlPaths.AdminOffer]: 'Offers',
  [UrlPaths.AdminPendingBets]: 'Settle Pending Picks',
  [UrlPaths.AdminUsersBalance]: 'Users Balance',
  [UrlPaths.AdminWeeklyBalance]: 'Weekly Balance',
  [UrlPaths.AdminPlayFreeBonusCredit]: 'Play Free Bonus Credit',
  [UrlPaths.AdminWithdrawable]: 'Manage Amount Available to Withdraw',
  [UrlPaths.AdminLineExposure]: 'Line Exposure',
  [UrlPaths.ManageWithdrawalOffer]: 'Manage Withdrawal Offer',
  [UrlPaths.SendSMS]: 'Send SMS',
  [UrlPaths.ManageAgentReferralCodes]: 'Manage Partner Referral Codes',
  [UrlPaths.PartnersReferralCodes]: 'Manage Partner Referral Codes',
  [UrlPaths.ManagePartnersPams]: 'Add/Edit Agents & PAMs',
  [UrlPaths.ManageUsers]: 'Add/Edit Users',
  [UrlPaths.AdminUserLimits]: 'Entry Limits',
};

export default function AdminBreadCrumb() {
  const { pathname } = useRouter();

  return (
    <Breadcrumbs
      aria-label="breadcrumb"
      className={'bg-gray-200 px-6 py-2'}
      separator={<NavigateNextIcon fontSize="small" />}
    >
      <Link
        underline="hover"
        sx={{ display: 'flex', alignItems: 'center' }}
        color="inherit"
        href="/admin"
      >
        <HomeIcon sx={{ mr: 0.5 }} fontSize="inherit" />
        Home
      </Link>
      {pathname !== UrlPaths.Admin && pathname !== UrlPaths.Partners && (
        <Link
          underline="hover"
          sx={{ display: 'flex', alignItems: 'center' }}
          color="inherit"
          href={pathname}
        >
          {breadcrumbNameMap[pathname] ||
            pathname
              .replace('/admin/', '')
              .replace('/partners/', '')
              .replace(/\b[a-z]/g, (x) => x.toUpperCase())
              .replaceAll('-', ' ')}
        </Link>
      )}
    </Breadcrumbs>
  );
}
