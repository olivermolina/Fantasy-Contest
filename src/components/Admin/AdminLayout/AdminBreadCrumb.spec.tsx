import React from 'react';
import { render } from '@testing-library/react';
import AdminBreadCrumb from './AdminBreadCrumb';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));
describe('AdminBreadCrumb', () => {
  it('should render', () => {
    const { container } = render(<AdminBreadCrumb />);
    expect(container).toMatchSnapshot();
  });
});
