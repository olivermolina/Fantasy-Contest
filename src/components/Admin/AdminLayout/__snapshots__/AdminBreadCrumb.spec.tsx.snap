// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminBreadCrumb should render 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root bg-gray-200 px-6 py-2 css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
  >
    <ol
      class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1f2xi7l-MuiTypography-root-MuiLink-root"
          href="/admin"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vnn4cn-MuiSvgIcon-root"
            data-testid="HomeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
          Home
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="NavigateNextIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1f2xi7l-MuiTypography-root-MuiLink-root"
          href=""
        />
      </li>
    </ol>
  </nav>
</div>
`;
